lib/Doctrine/Common/Cache/Psr6/DoctrineProvider.php

```diff
@@ -66,6 +70,33 @@ protected function doFetch($id)
         return $item->isHit() ? $item->get() : false;
     }
 
+    /**
+     * @inheritDoc
+     */
+    protected function doFetchMultiple(array $keys): array
+    {
+        $items = $this->pool->getItems(array_map('rawurlencode', $keys));
+        if ($items instanceof Traversable) {
+            $items = iterator_to_array($items);
+        }
+
+        $result = [];
+        foreach ($keys as $key) {
+            // To be checked whether redoing the rawurlencode is slower or faster than using a
+            // single `array_combine` to create a mapping of keys to encoded keys, or comparing
+            // that to building both the list and the map with a loop instead of the array_map
```

Comment:
I think we can drop this comment - if someone notices a performance complaint they'll identify the cause soon enough.
```suggestion
```